packages/http/src/metadata.ts

```diff
@@ -466,7 +495,10 @@ export function createMetadataInfo(program: Program, options?: MetadataInfoOptio
 
   function isOptional(property: ModelProperty, visibility: Visibility): boolean {
     // Properties are only made optional for update visibility
```

Comment:
I would update this to something like "Properties are made optional on patch requests if the visibility includes update, but not for array elements with item flag since you need to provide whole array elements with required properties even in a patch."